Create tests/decide_gi.py test method for is_balanced_or_is_bijected fixes #31

mvandermade opened the pull request:
Tests the function is_balanced_or_bijected.

The test structure is based on the tests/graph.py file.

Uses a dictionary to store the results from canvas.
Runs the test for each combination of graphs within a .grl file.
If the test fails for one graph combination debug information is printed.

Added a integer of the total number of expected test runs.
This number is displayed in the test summary.

mvandermade committed:
Create decide_gi.py

test based on the tests/graph.py file.
Uses a dictionary to store the results from canvas.
Runs the test for each combination of graphs within a .grl file.

VeraPrinsen commented on tests/decide_gi.py:
```diff
@@ -0,0 +1,101 @@
+graph_count = 0
+for file in files:
+    filename = '../test_graphs/color_refinement/' + file + '.grl'
+    with open(filename) as f:
```
In input_output/file_output (wat file_io moet heten besef ik me nu), zit een methode load_graph_list, waarin dit is geimplementeerd. Dan krijg je ook gelijk de L[0], zodat je dat hier nergens meer hoeft te doen.

VeraPrinsen commented on tests/decide_gi.py:
```diff
@@ -0,0 +1,101 @@
+                    print('---------------------------')
+                    print("Statistics of " + file + "-" + str(i) + "_" + str(j) + ":")
+                    print('---------------------------')
+                    print("Is calculated is_balanced_or_bijected correctly: " + str(is_balanced_or_bijected_test_result))
```
Je kunt hier om de gefaalde tests te benadrukken de fail() methode uit sys_output gebruiken die in input_output staat.

VeraPrinsen commented on tests/decide_gi.py:
```diff
@@ -0,0 +1,101 @@
+    test_is_balanced, test_is_bijected = is_balanced_or_bijected(graph)
+
+    if not (test_is_balanced == is_balanced and test_is_bijected == is_bijected):
+        print('DEBUG...')
```
Als dit voorkomt, wordt dit geprint voordat je weet over welke file het gaat.

VeraPrinsen commented on tests/decide_gi.py:
```diff
@@ -0,0 +1,101 @@
+#           Tuple containing : (is_balanced, is_bijected) ((bool, bool))
+
+expected_results = {}
+expected_results['colorref_smallexample_2_49'] = {0:
```
Vind de indentatie hier niet zo leesbaar, maar is meer een style dingetje

mvandermade committed:
Update decide_gi.py

Indentation of dict adjusted to meet PEP8
use of load_graph

mvandermade committed:
Update decide_gi.py

Throwing exception to catch error messages
Error messages formatted to match fail()
Also added:
 - dict() instead of {}
- validate the graph count against the total tests value.

JKleinRot commented on tests/decide_gi.py:
```diff
@@ -0,0 +1,93 @@
+                                                  }
+
+
+def test_balanced_or_bijected(graph: 'Graph', is_balanced: bool, is_bijected: bool):
```
Volgens mij moet de typing van een boolean ook met hoofdletter en '': `is_balanced: 'Bool'`

mvandermade replied:
Beide zijn goed volgens de pydocs, voor consistentie heb ik het aangepast naar 'Bool'.

https://docs.python.org/3/library/typing.html

JKleinRot commented on tests/decide_gi.py:
```diff
@@ -0,0 +1,93 @@
+
+def test_balanced_or_bijected(graph: 'Graph', is_balanced: bool, is_bijected: bool):
+    """
+    Tests the is_balanced_or_bijected method of decide_gi.py agains expected answers for is_balanced and is_bijected
```
against

JKleinRot commented on tests/decide_gi.py:
```diff
@@ -0,0 +1,93 @@
+from algorithms.color_refinement import color_refinement
+
+# Expected test results
+total_tests = 22
```
Dit vraagt wel om handmatig aanpassen van het aantal testen dat gerund wordt. Als je er een file bij toevoegt en je past dit niet aan, dan faalt de test als nog. Mag om mij wel blijven staan, maar volgens mij biedt het niet wat extra's in de test.

JKleinRot reviewed (changes requested):
Paar kleine dingen

mvandermade committed:
Update decide_gi.py

Typing of test_balanced_or_bijected bool -> 'Bool'

mvandermade committed:
Update decide_gi.py

typo
